android/src/main/java/cl/json/social/SingleShareIntent.java

```diff
@@ -39,9 +39,17 @@ public void open(ReadableMap options) throws ActivityNotFoundException {
                 System.out.println("NOT INSTALLED");
                 String url = "";
                 if(getDefaultWebLink() != null) {
-                    url = getDefaultWebLink()
-                            .replace("{url}",       this.urlEncode( options.getString("url") ) )
-                            .replace("{message}",   this.urlEncode( options.getString("message") ));
+                    // Check if default url has valid replacement strings.
+                    // Replace was failing on pixel phones when there was no match?!
+                    if(getDefaultWebLink().contains("{url}") && getDefaultWebLink().contains("{message}")) {
```

Comment:
It would probably be less awkward to set a base `url = getDefaultWebLink();` and then have two separate conditionals where if there's a URL, you replace URL, and likewise for message, instead of needing three separate cases.

Fundamentally, though, what is breaking about the replacement? If the link contains no `{message}`, for example, the replacement should do nothing and I don't see a problem.

Reply:
facebook and twitter urls looks like this: 

DEFAULT_WEB_LINK = "https://www.facebook.com/sharer/sharer.php?u={url}";
DEFAULT_WEB_LINK = "https://twitter.com/intent/tweet?text={message}&url={url}";

Yes.. for pixels it its causing an error and the match is failing when trying to find {message} for facebook when the url does not have it?!!  then the url becomes null. I think the ifs are still needed to avoid having the replace fail.

i have updated spacing a bit

Reply:
Ahh yeah.. its probablay failing the getString .. let me double check and update.

Reply:
@FFS-Evan Yes you correct. the null message is causing error. I can fix this in our react project by passing in an empty string, no need for this. Will close this and have new pr for other fix.